### PR TITLE
docs: add Godot 4 debugger description

### PR DESCRIPTION
**Pull request by xiliuya**
`dap-mode` is already support `GDScript/godot4`. So add a line of  description.

### CODE REVIEW

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
LGTM.

File: README.md [hunk 1]
```diff
@@ -297,7 +297,11 @@ Code example:
 
 ## Using the debugger
 
-Emacs GDScript mode includes support for the GDScript debugger. You can use breakpoints, use code stepping functions, see your nodes' state at runtime, and more.
+Emacs GDScript mode includes support for the GDScript debugger. 
+
+*The debugger provided in this package is mainly for Godot 3. Godot 4 has DAP support , and [dap-mode](https://github.com/emacs-lsp/dap-mode) is already support GDScript.*
```
Comment (NathanLovato) [inline README.md:302]:
```suggestion
*The debugger in this package is only for Godot 3. Godot 4 supports the Debugger Adapter Procol (DAP), which you can use with the [dap-mode](https://github.com/emacs-lsp/dap-mode) package.*
```

File: README.md [hunk 2]
```diff
@@ -297,7 +297,11 @@ Code example:
...
+
+You can use breakpoints, use code stepping functions, see your nodes' state at runtime, and more.
```
Comment (NathanLovato) [inline README.md:304]:
```suggestion
You can use the debugger tools to manage breakpoints, step through code, and more.
```

**Comment by NathanLovato** [comment]
When using acronyms like DAP in documentation, please make sure to at least expand it once. A general issue with Emacs packages' documentation is they tend to assume a lot of technical knowledge on the users' part, hurting the accessibility.

It's a good contribution though, let's get this merged indeed.

**Pull request merged**